addons/account/models/account_bank_statement.py

```diff
@@ -915,11 +925,29 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
             if abs(total)>0.00001:
                 partner_id = self.partner_id and self.partner_id.id or False
                 partner_type = False
-                if partner_id:
-                    if total < 0:
-                        partner_type = 'supplier'
-                    else:
-                        partner_type = 'customer'
+                # Check based on account type whether we have a payment for receivable (customer) or payable (supplier) accounts and fallback in case none of them are used
+                if partner_id and partner_id != self.env.user.company_id.partner_id:
+                    for account in partner_accounts:
+                        block = False
+                        if account.user_type_id.id == self.env.ref('account.data_account_type_payable').id:
```

Comment:
you could put `account_type_payable = self.env.ref('account.data_account_type_payable')` before the loop for, to improve the speed.

Reply:
@fmdl 
Jep, I thought about it, but was concentrated on the logical and accounting part...will do now...thx